ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -776,8 +777,12 @@ impl<Register: Send + Sync + 'static> TraceBuilder<Register> {
         // Check if this is a recursive call by scanning the call stack for the callee.
         let is_recursive = self.frames.iter().any(|f| f.funcidx == Some(*callee));
 
+        // FIXME: currently all functions marked idempotent are outlined, however we should be able
```

Comment:
I'm not sure this would be a good idea, because at this point in things we don't know what's actually constant or not -- we know *some* constant things, but the optimiser will find out more. Either option feels supoptimal, but inlining things is hard, so I'm inclined not to consider it. [And it would be an `OPT` not a `FIXME`, even if we did want to do it.]